tests/tests_suite_test.go

```diff
@@ -28,4 +28,12 @@ var _ = BeforeSuite(func() {
 })
 
 var _ = AfterSuite(func() {
+	client, err := tests.GetKubeClient()
+	tests.PanicOnError(err)
+
+	for k, v := range tests.Namespaces {
```

Comment:
why not just iterate through a list of all the namespaces in the cluster and delete the ones that have 'cdi-test' as a prefix?

This would ensure that we also cleanup any namespaces that got leaked from a previous test run that was aborted early.

Reply:
@davidvossel yes that works fine. We'd want to make sure we're not deleting a ns that is still being used though. In my head the ns is in a test global struct and deleting it would be part of the cleanup done in an `AfterSuite`.

Reply:
@jeffvance as this code is in AfterSuite, that would not be a problem. I took a stab at something to clean up namespaces, of course just iterating over the existing name spaces is much simpler.

Reply:
my point is, what if AfterSuite doesn't run?

It's possible to terminate the tests and not let AfterSuite complete. At that point we'd have namespaces that we'd have no reference to because the state (the map in this case) that tracked the namespaces would not be recoverable.

As long as we only run the destroy all function at the beginning and end of the test suite, nothing should be using the namespaces in question.

Reply:
Doesn't gingko guarantee that AfterSuite is run (even when ctrl-c is used), unless of course you hard kill the process, but at that point we can't give many guarantees at all. But I see your point about running it in BeforeSuite as well, just to remove any stuff that was left over from a hard kill.

Reply:
@davidvossel I think an `AfterSuite` will run as reliably as a `defer`, so other than a node crash it should run.
I think I'm misunderstanding this ns list and the idea of deleting namespaces in a `BeforeSuite`. If the namespaces are random, and if the ns list is global to a test suite,  (and if the list is empty when the test_suite first runs,) then how can a `BeforeSuite` delete the contents of this list?

Reply:
> unless of course you hard kill the process,

right, that's what I'm getting at. There's going to be a point during development where that occurs. Maybe the testsuite crashes, maybe something hangs and we have to force kill the test. It's not common, but it does occur regularly enough that it would be nice if subsequent test runs know how to start from a clean slate.